internal/compiler/widgets/fluent-base/widget-table-view.slint

```diff
@@ -0,0 +1,179 @@
+    Rectangle {  
+        y: parent.height - self.height;
+        width: 100%;
+        height: 1px;
+        background: Palette.neutralPrimaryAlt;
+    }
+}
+
+component TableViewRow inherits Rectangle {
+    min-height: 42px;
+
+    touch-area := TouchArea {}
+
+    HorizontalLayout {  
+       @children
+    }
+
+    states [
+        hover when touch-area.has-hover : {
+            background: Palette.neutralLighter;
+        }
+    ]
+}
+
+export component StandardTableView inherits Rectangle {
+    callback sort-ascending(int);
+    callback sort-descending(int);
+    
+    out property <int> current-sort-column: -1;
+    in-out property <[TableColumn]> columns;
+    in property <[[StandardListViewItem]]> rows;
+    in property <length> min-header-height: 42px;
+   
+    min-width: 400px;
+    min-height: 200px;
+    horizontal-stretch: 1;
+    vertical-stretch: 1;
+
+    function sort(index: int) {
+        if (root.current-sort-column != index) {
+            root.columns[root.current-sort-column].sort-order = SortOrder.unsorted;
+        }
+
+        if(root.columns[index].sort-order == SortOrder.ascending) {
+            root.columns[index].sort-order = SortOrder.descending;
+            root.sort-descending(index);
+        } else {
+            root.columns[index].sort-order = SortOrder.ascending;
+            root.sort-ascending(index);
+        }
+
+        root.current-sort-column = index;
+    }
+
+    VerticalLayout {  
+        HorizontalLayout {  
+            padding-right: 20px;
+            min-height: root.min-header-height;
+            vertical-stretch: 0;
+            for column[index] in root.columns : TableViewColumn {
+                sort-order: column.sort-order;
+                min-width: column.min-width;
+                horizontal-stretch: column.horizontal-stretch;
+
+                Text {  
+                    vertical-alignment: center;
+                    horizontal-stretch: column.horizontal-stretch;
+                    min-width: column.min-width;
+                    text: column.title;
+                    font-weight: 900;
+                }
+                
+                clicked => {  
+                    root.sort(index);
+                }
+            }
+        }
+
+        scroll-view := ScrollView {  
+            border-width: 0;
+            vertical-stretch: 1;
+
+            VerticalLayout {  
+                alignment: start;
+
+                for row in root.rows : TableViewRow {
+                    for cell[index] in row : TableViewCell {
+                        private property <bool> has_inner_focus;
+                        horizontal-stretch: root.columns[index].horizontal-stretch;
+                        min-width: root.columns[index].min-width;
+
+                        Rectangle {  
+                            // height: 100%;
+                            line-edit := LineEditInner {
+                                opacity: self.has-focus ? 1.0 : 0.0;
```

Comment:
I'm curious, is `opacity` used here instead of `visible` because it can be animated?

Reply:
It was the idea but because there is no animation at the moment I will change it to visible

Reply:
It's because I need to focus the LineEdit what is not possible if it is hidden.